src/js/base/fetch.js

```diff
@@ -48,59 +118,52 @@ function getUrl(url, data) {
 }
 
 export function getData(response, dataType) {
+  response = response.clone();
+
   if (dataType === 'json' && response.status !== 204) return response.json();
+
   return response.text();
 }
 
-export function handleJSON(response) {
-  if (!response.ok) return Promise.reject(response);
+export async function handleJSON(response) {
+  if (!response) return;
 
-  return getData(response, 'json');
-}
-
-export default async(url, opts) => {
-  const token = await Radio.request('auth', 'getToken');
+  const responseData = await getData(response, 'json');
 
-  const options = extend({}, opts);
+  if (!response.ok) return Promise.reject({ response, responseData });
 
-  if (!options.method || options.method === 'GET') {
-    url = getUrl(url, options.data);
-  } else if (options.data) {
-    options.body = options.data;
-  }
+  return responseData;
+}
 
-  defaults(options, {
-    dataType: 'json',
-    headers: defaults(options.headers, {
-      'Accept': 'application/vnd.api+json',
-      'Content-Type': 'application/vnd.api+json',
-    }),
-  });
+export function handleError(error) {
+  if (error.name !== 'AbortError') throw error;
+}
 
-  if (token) options.headers.Authorization = `Bearer ${ token }`;
+export default async(url, options) => {
+  const fetcher = getActiveFetcher(url, options) || buildFetcher(url, options);
 
-  // Attach preferred workspace to request
-  const currentWorkspace = Radio.request('bootstrap', 'currentWorkspace');
-  if (currentWorkspace) options.headers.Workspace = currentWorkspace.id;
+  return fetcher
+    .then(async response => {
+      removeFetcher(url);
 
-  return fetch(url, options)
-    .then(response => {
       if (!response.ok) {
-        if (response.status >= 400) {
-          logResponse(url, options, response.clone());
-        }
+        const responseClone = response.clone();
 
         if (response.status === 401) {
-          Radio.request('auth', 'logout');
-          return Promise.reject(response);
+          const responseData = await responseClone.json();
+          if (get(responseData, ['errors', 0, 'code']) !== '5000') Radio.request('auth', 'logout');
```

Comment:
with the exception of 1 401 we currently can receive, we want to logout at this point.